### PR TITLE
Add Pet Mod of the Season

### PR DESCRIPTION
**Pull request by KrisXV**

### CODE REVIEW

**Comment by Zarel** [comment]
Was the modified Species Clause removed? I don't see it in the diff.

**Comment by KrisXV** [comment]
Yeah, i changed the dex numbers in the mod file so there's no need to worry about it @Zarel

**Comment by Zarel** [comment]
This will still lead to client bugs. Can you just ask them to choose different mons than Rotom?

**Comment by KrisXV** [comment]
No..., if it will lead to client bugs ill leave the numbers as normal and just modify species clause again

**Comment by Zarel** [comment]
You'd need to call it something besides "Species Clause".

Specifically, client assumes that if there exists a rule named "Species Clause", there will be at most one Pokémon for every dex number.

**Comment by KrisXV** [comment]
Does this work @Zarel ?

I didn't add it to the CUSTOM-RULES.md documentation because it wont work outside of this format

File: data/mods/petmod/rulesets.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+/**@type {{[k: string]: ModdedFormatsData}} */
+let BattleFormats = {
+	standardpetmod: {
+		inherit: true,
+		ruleset: ['[Gen 7] OU', '!Species Clause'],
+		banlist: ['All Pokemon'],
+		onBegin() {
+			// The only validator rule this currently modifies is Species Clause, so the added rule is just this
+			this.add('rule', 'Species Clause: Limit one of each Pokémon');
```
Comment (Zarel) [inline data/mods/petmod/rulesets.js]: No, this doesn't work. This is the line client has trouble with. You can't send a message starting with `|rule|Species Clause:`.

Comment (KrisXV) [reply data/mods/petmod/rulesets.js]: ok

File: data/mods/petmod/rulesets.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+/**@type {{[k: string]: ModdedFormatsData}} */
+let BattleFormats = {
+	standardpetmod: {
+		inherit: true,
+		ruleset: ['[Gen 7] OU', '!Species Clause'],
+		banlist: ['All Pokemon'],
+		onBegin() {
+			// The only validator rule this currently modifies is Species Clause, so the added rule is just this
+			this.add('rule', 'PM Species Clause: Limit one of each Pokémon');
```
Comment (Zarel) [inline data/mods/petmod/rulesets.js]: Why not Clean Slate? Will all pet mods have the same species clause?

Comment (scoopapa) [reply data/mods/petmod/rulesets.js]: No, all pet mods do not have the same species clause, it's just this one that needs the modified one.

Comment (Zarel) [reply data/mods/petmod/rulesets.js]: Then it should not called "PM". Also the mod should not be called "petmod".

Comment (KrisXV) [reply data/mods/petmod/rulesets.js]: I’m calling it petmod for ease of client access for future pet mods

Comment (KrisXV) [reply data/mods/petmod/rulesets.js]: anyways, fixed

**Pull request merged**